designer/client/conditions/select-condition-helpers.ts

```diff
@@ -1,57 +1,27 @@
 import { ConditionData } from "./SelectConditions";
+import { ConditionRef } from "@xgovformbuilder/model";
 
-export const isObjectCondition = (condition: ConditionData) => {
-  return typeof condition.value !== "string";
-};
-
-export const isStringCondition = (condition: ConditionData) => {
+export function isStringCondition(condition: ConditionData) {
   return typeof condition.value === "string";
-};
+}
 
-export const hasConditionName = (condition: any) => {
-  return !!condition?.conditionName;
-};
+export function isConditionRef(condition): condition is ConditionRef {
+  return condition?.conditionName !== undefined;
+}
 
-export const hasNestedCondition = (condition: ConditionData) => {
+export function hasNestedCondition(condition: ConditionData) {
   if (typeof condition.value === "string") {
     return false;
   }
-  return condition.value.conditions?.find?.(hasConditionName) ?? false;
-};
-
-export const isDuplicateCondition = (
-  conditions: any[],
-  conditionName: string
-) => {
-  return !!conditions.find((condition) => condition.name === conditionName);
-};
-
-export const getFieldNameSubstring = (sectionFieldName: string) => {
-  return sectionFieldName.substring(sectionFieldName.indexOf("."));
-};
-
-export function conditionsByType(conditions: ConditionData[]) {
-  return conditions.reduce(
-    (conditionsByType, currentValue) => {
-      if (isStringCondition(currentValue)) {
-        conditionsByType.string.push(currentValue);
-      } else if (!!hasNestedCondition(currentValue)) {
-        conditionsByType.nested.push(currentValue);
-      } else if (isObjectCondition(currentValue)) {
-        conditionsByType.object.push(currentValue);
-      }
-      return conditionsByType;
-    },
-    {
-      string: [],
-      nested: [],
-      object: [],
-    } as ConditionByTypeMap
-  );
+  return condition.value.conditions?.find?.(isConditionRef) ?? false;
 }
 
-type ConditionByTypeMap = {
-  string: ConditionData[];
-  nested: ConditionData[];
-  object: ConditionData[];
-};
+export function getConditionType(condition: ConditionData) {
+  if (isStringCondition(condition)) {
+    return "string";
+  }
+  if (!!hasNestedCondition(condition)) {
```

Comment:
if you use a falsey value in an if statement, it will evaluate to false. So I'm not sure if you need `!!`. If `hasNestedCondition` returns undefined - this if statement evals to false. 

`hasNestedCondition` should probably just return a boolean too - or change the name of the function if it's returning a nested condition. `getNestedCondition`